### ISSUE TITLE
Multiple of four spaces

### ISSUE THREAD

**Issue by thisirs** [wbolster/black-macchiato#11]
Hi,

I'm using `black-macchiato` to format python code in reStructuredText code chunks. Unfortunately that code is not always indented by a multiple a four spaces and an exception is raised. Would it be possible to allow it? Looking at the code, the multiple four spaces are required to be able to prepend some `if True:` to have legit code for `black`. Is there a reason not to dedent that code first and reindent it at the end?

Thanks,

**Comment by wbolster** [comment]
yes there is: it affects the line length and hence wrapping behaviour

**Comment by wbolster** [comment]
btw, code blocks in rst files can be indented by 4 spaces without problem

**Comment by thisirs** [comment]
> yes there is: it affects the line length and hence wrapping behavior

Indeed! But only for code chunks taken from plain python file. Do we want to have that wrapping behavior when reformatting a deeply nested code block in some markdown file? Maybe not.


> btw, code blocks in rst files can be indented by 4 spaces without problem

Absolutely not a rst expert. I happen to have 3 spaces which can probably be rewritten to have 4. But the problem remains for any chunk of code taken in a plain text file that has no reason to be 4 spaces indented. 

A solution would be to pad each line to have a multiple of 4 spaces and then remove the padding. But this might break here strings... Another way is to add a cli option to dedent first...

**Comment by wbolster** [comment]
yep, i understand the problem. ideally things like indented code blocks in a `.rst` file are formatted such that `.rst` related indents (but not python related indents) are ignored.

not sure how this can be achieved though in a generic way. transparent automatic indent detection etc would format indented lines of code in a regular python file using the wrong line length. adding custom command line flags may not be ideal either; `black-macchiato` currently takes no arguments and forwards everything to `black` (by design).
